src/Plugin/Commerce/TaxType/TaxCloud.php

```diff
@@ -286,7 +288,7 @@ public function apply(OrderInterface $order) {
           $order_item_total_tax_amount = $order_item->getAdjustedTotalPrice()->multiply($percentage);
           $order_item_tax_amount = $order_item->getAdjustedUnitPrice()->multiply($percentage);
 
-          if ($this->shouldRound()) {
+          if ($this->shouldRound() || $rounding_method == 'per_line') {
```

Comment:
It doesn't look fine for me. Since we have two rounding settings.
Maybe we need to remove this setting: round_tax_amount, since we have another setting to round it per order line, total.

Also I think we might need the third option:
1. Do not round
2. Round per order line
3. Round per total

I think you need to identify how taxcloud rounds taxes (per order line or total). I think it's total. But please check a response from taxcloud to make sure.

Then, for example:
If tax cloud rounding method == total and tax_rounding_method == total
  do nothing
if tax cloud rounding method == total and tax_rounding_method == per_line
  Round per order line
if tax cloud rounding method == total and tax_rounding_method == do not round
  Make sure we don't round tax amount